File: Codigos estudiantes por lenguaje/Java/Daniel Cangrejo/Algoritmos_de_Busqueda/NumMissing.java

public class Busquedas1 {
//BUSQUEDA DE UN NUMERO PERDIDO
	
	static int getMissingNo(int a[], int n) {
		
		int x1 = a[0];
		int x2 = 1;
		for (int i = 1; i < n; i++) {
			x1^=a[i];
			System.out.println(i + " -> " + x1);
		}
		for (int i = 2; i <= n+1; i++) {
			x2^=i;
			System.out.println(i + " -> " + x2);
		}
		return x1^x2;
	}
	
	
	public static void main(String[] args) {
		int [] arr = {2,3,4,5,6,7,8};
		int missed = getMissingNo(arr, arr.length);
		System.out.println("El numero que falta es: " + missed);
 
	}

}

File: Codigos estudiantes por lenguaje/PY/Daniel Cangrejo/Busquedas/NumMissing.py
from sys import stdin, stdout
rl = stdin.readline
wr = stdout.write


def get_Missing_N(a=list, n=int):
    x1 = a[0]
    x2 = 1
    for i in range(1, n):  # Hash de Datos que tengo
        x1 ^= a[i]
        # wr(f'{i} -> {x1}\n')  # Print Statement
    for i in range(2, n+2):  # Hash de Datos que debería tener
        x2 ^= i
        # wr(f'{i} -> {x2}\n')  # Print Statement
    return x1 ^ x2


arr = list(map(int, rl().split()))
missed = get_Missing_N(arr, len(arr))
wr(f'El numero que falta es: {missed}')
